libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/ImageEditor.kt

```diff
@@ -52,22 +57,56 @@ class ImageEditor private constructor(
         loadIntoImageView.invoke(imageUrl, imageView, scaleType)
     }
 
+    fun onEditorAction(action: EditorAction) {
+        onEditorAction.invoke(action)
+        if (action is EditorCancelled || action is EditorFinishedEditing) {
+            actions.clear()
+        }
+    }
+
+    sealed class EditorAction {
+        // General actions
+        data class EditorShown(val numOfImages: Int) : EditorAction()
+        object EditorCancelled : EditorAction()
+        data class EditorFinishedEditing(val outputDataList: List<OutputData>, val actions: List<Action>) :
+                EditorAction()
+
+        // Preview screen actions
+        data class PreviewImageSelected(val highResImageUrl: String, val selectedPosition: Int) : EditorAction()
+        data class PreviewInsertImagesClicked(val outputDataList: List<OutputData>) : EditorAction()
+        object PreviewCropMenuClicked : EditorAction()
+
+        // Crop screen actions
+        object CropOpened : EditorAction()
+        data class CropDoneMenuClicked(val outputData: OutputData) : EditorAction()
+        data class CropSuccessful(val cropResult: CropResult) : EditorAction()
+    }
+
+    sealed class Action(val label: String) {
+        object Crop : Action("crop")
+    }
+
     companion object {
         private lateinit var INSTANCE: ImageEditor
 
         val instance: ImageEditor get() = INSTANCE
 
+        // The actions that were made in this session.
```

Comment:
I'm wondering if we could move the responsibility of keeping track of the actions to the client. We already emit `CropSuccessful` so the client can easily track the actions, right? Wdyt?

Reply:
Done here: 4760c7f